src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php

```diff
@@ -100,30 +169,37 @@ private function isInlineableDefinition($id, Definition $definition, ServiceRefe
             return false;
         }
 
-        if (!$graph->hasNode($id)) {
+        if (!$this->graph->hasNode($id)) {
             return true;
         }
 
         if ($this->currentId == $id) {
             return false;
         }
+        $this->connectedIds[$id] = true;
 
-        $ids = array();
-        foreach ($graph->getNode($id)->getInEdges() as $edge) {
+        $srcIds = array();
+        $srcIdsCount = 0;
+        foreach ($this->graph->getNode($id)->getInEdges() as $edge) {
+            $srcId = $edge->getSourceNode()->getId();
+            $this->connectedIds[$srcId] = true;
             if ($edge->isWeak()) {
```

Comment:
if an edge is weak, other edges will not be marked as connected here. Is is expected ?